source/language/delays.rst

```diff
@@ -127,23 +128,43 @@ these side effects. Also contrary to TeX, we prohibit overlapping gates.
 Operations on durations
 -----------------------
 
-We can add/subtract two durations, or multiply or divide them by a constant, to get a new
-duration. Division of two durations results in a machine-precision float 
-(see :ref:`divideDuration`). Negative durations are allowed, however
-passing a negative duration to a ``gate[duration]`` or ``box[duration]`` expression will result in an error.
-All operations on durations happen at compile time since ultimately all
-durations, including stretches, will be resolved to constants.
+We can add/subtract two durations, or multiply or divide them by a
+constant, to get a new duration. Division of two durations results in
+a machine-precision float (see :ref:`divideDuration`). Negative
+durations are allowed, however passing a negative duration to a
+``gate[duration]`` or ``box[duration]`` expression will result in an
+error.  All operations on ``const`` durations, including stretches,
+can happen at compile time. Note that some programs may make use of
+non-const duration (e.g. as ``input`` parameters or in
+backend-specific externs).
 
 .. code-block::
 
-       duration a = 300ns;
-       duration b = durationof({x $0;});
+       const duration a = 300ns;
+       const duration b = durationof({x $0;});
        stretch c;
        // stretchy duration with min=300ns
        stretch d = a + 2 * c;
        // stretchy duration with backtracking by up to half b
        stretch e = -0.5 * b + c;
 
+Because `stretch` introduces a `const duration`, initialization with a non-`const` expression are a type error. On the other hand, because stretch resolution requires solving a system of constraints, a compiler may reject a well-typed program involving stretch durations due to a failure to resolve the constraint system.
```

Comment:
Imo we shouldn't refer to "initialisation" of a `stretch`, because the `stretch a = <expr>;` statement doesn't _set_ `a` to the value of `<expr>`, it just introduces another lower-bound on the value of `a`.

I'd prefer
> Because `stretch` introduces a `const duration`, the optional lower bound express must also have type `const duration`.

(or something along those lines).